osu.Game/Rulesets/Objects/Pooling/DrawableObject.cs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable enable
+
+using System.Diagnostics;
+using osu.Framework.Graphics.Performance;
+using osu.Framework.Graphics.Pooling;
+
+namespace osu.Game.Rulesets.Objects.Pooling
+{
+    /// <summary>
+    /// A <see cref="PoolableDrawable"/> that is controlled by <see cref="Entry"/> to implement drawable pooling and replay rewinding.
+    /// </summary>
+    /// <typeparam name="TEntry">The <see cref="LifetimeEntry"/> type storing state and controlling this drawable.</typeparam>
+    public abstract class DrawableObject<TEntry> : PoolableDrawable where TEntry : LifetimeEntry
```

Comment:
Needs a better name. `DrawableObject` is too general for what this is. `DrawableObjectWithLifetime`, maybe. Or `PoolableDrawableWithLifetime`...?

This is generic enough that you could probably even put it in framework at this rate.

Reply:
I like `PoolableDrawableWithLifetime`. Hopefully this class can be used for other things like followpoints?